packages/grant-explorer/src/features/projects/hooks/useProject.ts

```diff
@@ -0,0 +1,18 @@
+import useSWR from "swr";
+import { DataLayer } from "data-layer";
+import { AlloVersion } from "data-layer/src/data-layer.types";
+
+type Params = {
+  projectId?: string;
+};
+
+export function useProject(params: Params, dataLayer: DataLayer) {
+  const shouldFetch = Object.values(params).every(Boolean);
```

Comment:
instead of `Boolean` we can explicitly check that the value is not `undefined` with `every((v) => v !== undefined)`